### PR TITLE
Premium mss activation [MAILPOET-2431]

### PR DESCRIPTION
**Pull request by JanJakes**
[MAILPOET-2431]

[MAILPOET-2431]: https://mailpoet.atlassian.net/browse/MAILPOET-2431

### CODE REVIEW

File: assets/js/src/settings/premium_tab/premium_tab.jsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            <div>
+              <input
+                type="text"
+                id="mailpoet_premium_key"
+                className="regular-text"
+                name="premium[premium_key]"
+                value={key || ''}
+                onChange={(event) => setKey(event.target.value.trim() || null)}
+              />
+              {' '}
```
[the remark this answers is not in the document]
Comment (JanJakes) [reply assets/js/src/settings/premium_tab/premium_tab.jsx]: Fixed.

File: assets/js/src/settings/premium_tab/messages/mss_messages.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import MailPoet from 'mailpoet';
+
+const validMessage = (
+  <div className="mailpoet_success">
+    {MailPoet.I18n.t('premiumTabMssValidMessage')}
+  </div>
+);
```
[the remark this answers is not in the document]
Comment (JanJakes) [reply assets/js/src/settings/premium_tab/messages/mss_messages.jsx]: Fixed.

File: assets/js/src/settings/premium_tab/messages/premium_messages.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import MailPoet from 'mailpoet';
+
+const validMessage = (
```
[the remark this answers is not in the document]
Comment (JanJakes) [reply assets/js/src/settings/premium_tab/messages/premium_messages.jsx]: Fixed.

File: assets/js/src/settings/premium_tab/messages/premium_messages.jsx [hunk 4]
```diff
@@ -2,32 +2,102 @@ import PropTypes from 'prop-types';
...
-    {MailPoet.I18n.t('premiumTabPremiumValidMessage')}
+    {MailPoet.I18n.t('premiumTabPremiumInstallingMessage')}
+  </div>
+);
+
+const activatingMessage = (
+  <div className="mailpoet_success">
+    {MailPoet.I18n.t('premiumTabPremiumActivatingMessage')}
+  </div>
+);
```
[the remark this answers is not in the document]
Comment (JanJakes) [reply assets/js/src/settings/premium_tab/messages/premium_messages.jsx]: Fixed.

**Comment by costasovo** [comment]
@codemonkey-jack Please check also other settings tabs during QA, this change contains a refactor that relates to all tabs.

**Comment by codemonkey-jack** [comment]
Works fine for me, now with Quentin in product QA.

**Comment by codemonkey-jack** [comment]
All changes look good, back with Quentin in product QA.

**Comment by JanJakes** [comment]
@costasovo Rebased and added 1 commit.

**Comment by codemonkey-jack** [comment]
Looks good to me, back with Quentin in product QA.

**Comment by codemonkey-jack** [comment]
@JanJakes Two checks tests failed, could you check when you get a chance, please? Thanks!

**Pull request merged**